db/compaction/compaction_iterator.cc

```diff
@@ -225,6 +230,13 @@ bool CompactionIterator::InvokeFilterIfNeeded(bool* need_skip,
           compaction_filter_skip_until_.rep());
       if (CompactionFilter::Decision::kUndetermined == filter &&
           !compaction_filter_->IsStackedBlobDbInternalCompactionFilter()) {
+        if (compaction_ == nullptr) {
```

Comment:
This assumes that flush won't trigger compaction filter, something we have considered changing?

Reply:
That check was already there; I just moved it up by ~10 lines.

It's OK to have this check (and it actually could be an assertion IMHO) because when using the integrated BlobDB, it is the flush process that extracts large values into blob files, which means a compaction filter used during flush cannot possibly encounter a blob index.